lib/go-atscfg/astatsdotconfig.go

```diff
@@ -47,7 +47,7 @@ func MakeAStatsDotConfig(
 	}
 	warnings := []string{}
 
-	if server.Profile == nil {
+	if server.ProfileNames == nil && len(server.ProfileNames) == 0 {
```

Comment:
This `len` check will never be used in a non-tautological way. If the length is 0 but it's not nil, the first term is `false` so it `&&` anything can't be true and the second term isn't checked. If it's `nil`, the length is always zero because `len` on a `nil` slice returns `0`.

This should just check the length.